app/src/main/java/com/example/root/smsreader/MainActivity.java

```diff
@@ -22,117 +15,116 @@
 
 public class MainActivity extends AppCompatActivity {
 
-public  int len_all_sms = 0;
-public int Count=0;
+    private static final String TAG = MainActivity.class.getSimpleName();
+    // Member variables have prefix "m". rest of the name uses CamelCase
+public  int mLenAllSms = 0;
+public int mCount = 0;
 
 
-    public String INR[] ;
-    ListView L1 ;
-    TextView count;
-    Button analysis;
+    public String mINR[] ;
+    ListView mLvSms;
+    TextView mTvCount; // for TextView, add Tv after prefix of "m"
+    Button mBtnAnalysis;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        
         setContentView(R.layout.activity_main);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
-        Log.v("Test", "List view initialized");
-        List<String> s;
-        s = readSms();
-        len_all_sms = s.size();
-        Log.v("Test", "readSms func called");
-        Log.v("Test", " "+ len_all_sms +" objects will be created now .....");
+        Log.d(TAG, "List view initialized"); //use Log.d() . this is not run during in release builds. Logging affects performance
+        List<String> smsList;
+        smsList = readSms();
+        mLenAllSms = smsList.size();
+        Log.d(TAG, "readSms func called");
+        Log.d(TAG, " "+ mLenAllSms +" objects will be created now .....");
 
-        SmsAttributes[] smsAttributes = new SmsAttributes[len_all_sms];
-        Log.v("Test", len_all_sms+" Objects created successfully...");
+        SmsAttributes[] smsAttributes = new SmsAttributes[mLenAllSms];
+        Log.d(TAG, mLenAllSms +" Objects created successfully...");
 
         try {
-    //L1 = (ListView) findViewById(R.id.listView);
-    count = (TextView)findViewById(R.id.textView3);
-
-
-
-
-
-
-    String messagecsv = "";
-
-    Log.v("Test", "going inside a for loop");
-
-    for(int index =0 ; index < s.size(); index++){
-        Log.v("Test", "In the for loop to call d class with index "+index);
+            //mLvSms = (ListView) findViewById(R.id.listView);
+            mTvCount = (TextView)findViewById(R.id.textView3);
+        
+            String messageCsv = ""; // use camelCase for naming variables
+        
+            Log.d(TAG, "going inside a for loop");
+        
+            for(int smsIndex =0 ; smsIndex < smsList.size(); smsIndex++){
```

Comment:
The index in a for loop should also have proper naming. While reviewing code after some days, you will not remember what exactly is happening. Making the variable names self explanatory is crucial.